src/io_uring/src/lib.rs

```diff
@@ -142,13 +143,15 @@ impl IoUring {
 
         let squeue = SubmissionQueue::new(fd, &params).map_err(Error::SQueue)?;
         let cqueue = CompletionQueue::new(fd, &params).map_err(Error::CQueue)?;
+        let slab = slab::Slab::with_capacity((params.sq_entries + params.cq_entries) as usize);
```

Comment:
check for overflow

Reply:
Maybe there is no need for checking overflow here.
We have do syscall with the same parameters and it succeed. So sq_entries + cq_entries will never overflow usize. Otherwise the previous syscall will return ENOMEM.
I added some comment about it.

Reply:
>We have do syscall with the same parameters and it succeed

It's not about overflowing usize in this case, because you only cast to `usize` after the addition, it's about overflowing `u32`.

So I suggest you either add overflow check or just cast both values to usize before doing the addition (with `into()` instead of `as`)